lib/members_page.rb

```diff
@@ -0,0 +1,72 @@
+# encoding: utf-8
+# frozen_string_literal: true
+require 'scraped'
+require_relative './decorators/party_name_attributes'
+
+class MembersPage < Scraped::HTML
+  decorator Scraped::Response::Decorator::AbsoluteUrls
+  decorator PartyNameAttributes
+
+  field :party_groupings do
+    trs = noko.xpath('//td[contains(.,"Distrito / Circunscripción")]/table/tr')
+    rows = trs.xpath('td[contains(.,"Circunscripción")]/parent::tr')
+
+    rows.map do |row|
+      fragment row => PartyGroupings
+    end
+  end
+
+  class PartyGroupings < Scraped::HTML
+    field :name do
+      party_image_node[:party_name]
+    end
+
+    field :id do
+      party_image_node[:party_id]
+    end
+
+    field :members do
+      noko.xpath('following-sibling::tr').slice_before { |i| i.text == noko.text }
+          .first.map do |row|
+        span = row.at_xpath('./td/@colspan')
+        next if span.to_s == '3'
```

Comment:
If you really need to do this sort of check it's much better to `.reject` things before calling `.map`, rather than `next`-ing out and then removing the blanks. But, again, I'm suspicious.